map-client/src/App.vue

```diff
@@ -226,29 +249,39 @@ export default {
       return `https://www.facebook.com/sharer.php?u=${item.url}`
     },
     clipDesc(desc) {
-      const clipped = desc.replace(/&\w+;/g, ' ')
-      return clipped.length > MAX_DESC_LENGTH
-        ? clipped.slice(0, MAX_DESC_LENGTH) + '…'
-        : clipped
+      const sanitized = desc.replace(/&\w+;/g, ' ')
+      return sanitized.length > MAX_DESC_LENGTH
+        ? sanitized.slice(0, MAX_DESC_LENGTH) + '…'
+        : sanitized
+    },
+    infiniteHandler($state) {
+      this.appendItems()
+
+      if (this.displayItems.length > 0) $state.loaded()
+      if (this.displayItems.length >= this.filteredItems.length) $state.complete()
     },
-    updateFilteredItems() {
-      this.loadingID = setTimeout(() => {
-        if (this.isSearchTypeString) {
-          this.filteredItems = this.searchString
-            ? this.items.filter(i => this.isMatchPattern(i))
-            : this.items
-        } else {
-          this.filteredItems = this.selectedPref
-            ? this.items.filter(i => this.isSelectedPref(i))
-            : this.items
-        }
-        this.loadingID = 0
-      }, 0)
+    initDisplayItems() {
+      this.page = 0
+      this.displayItems = []
+      this.appendItems()
+    },
+    appendItems() {
+      const index = this.page * this.perPage
+      if (this.filteredItems.length > index) {
+        const slice = this.filteredItems.slice(index, index + this.perPage)
+        this.displayItems.push(...slice)
+        this.page += 1
```

Comment:
[IMO]
細かいところですが、pushで直接 `this.displayItems` の値を変更するのではなく値を代入する方が読みやすくなおかつ安全かなと思いました！
```suggestion
        const nextItems = this.filteredItems.slice(index, index + this.perPage)
        this.displayItems = [...this.displayItems, ...nextItems]
        this.page += 1
```

Reply:
@chrome-cgi すみません、こちらの返事をいただき次第、この PR を master に入れてしまおうと思ってます。どうでしょう？

Reply:
返事遅れてすみません。

今回のように配列の末尾に短い配列を追加していくような場合、非破壊的な操作 ( `Array.concat` や spread operator を使った再代入など ) でやると毎度配列のコピーが発生するため、配列の長さが増えるほど性能に桁違いに差が出てきます。

参考ベンチマーク: https://www.measurethat.net/Benchmarks/ShowResult/107091

なのでここは `Array.push` のままで良いかと思います。

Reply:
返信ありがとうございます！おっしゃる通りコピー分重くなってしますね...問題ないと思います！